src/Bicep.Core/Configuration/bicepconfig.json

```diff
@@ -32,6 +32,14 @@
       }
     }
   },
+  "providerAliases": {
+    "br": {
+      "public": {
+        "registry": "mcr.microsoft.com",
+        "providerPath": "bicep/providers"

```

Comment:
Are there any issues with these being a subdir of the module alias? Thinking for example:
- Interfering with completions for modules
- Being able to permission "push az provider" separately from "push any bicep module".

Reply:
I don't understand this comment, what are you referring to by 'subdir' ?
In case you meant `namespaces` as defined in https://learn.microsoft.com/en-us/azure/container-registry/container-registry-best-practices#repository-namespaces, (per documentation)
> the registry manages all repositories independently, not as a hierarchy

Reply:
Thanks for clarifying - I was being lazy with terminology. I don't think you've answered the question though.

Reply:
I think Anthony's question here is about what happens when a module gets published to `bicep/providers/az`. We wouldn't do that intentionally, but we may need to take steps to ensure it's difficult to do by accident.

Reply:
Yeah it's a good point, I believe that the permissions are handled using scope maps - https://learn.microsoft.com/en-us/azure/container-registry/container-registry-repository-scoped-permissions?source=recommendations#manage-tokens-and-scope-maps.

I don't know what is the current scope for the credential used to publish modules. We would need to be careful of course to use it properly in case it has full access to the registry. Maybe we can restrict the scope somehow but it's not something that I have spent a lot of time researching.

This is generally a good question in terms of how we want to store the artifacts. The provider author , assuming is using the same ACR to store both modules and providers needs to manage this.

The tradeoff is that managing separate ACRs may add complexity to the managed infra so I chose to go in the path of least resistance  since we already have a registry deployed. If this is a blocking worry we could use a separate acr to host the az artifacts.

Reply:
Also, the same problem exists for modules regardless. What prevents an unintended overriding of a module ? Since all modules are stored in the same registry and published centrally, there is always a chance to mistakenly publish into the wrong repo. What kind of guardrails exist for this?

Reply:
We should probably talk about this in more detail, however is it a blocking issue for this PR?

@shenglol, do you mind providing your thoughts on this? I would like to really get an opportunity to estimate the investment and ROI of managing a separate registry.

Reply:
> Are there any issues with these being a subdir of the module alias? Thinking for example:
> 
> * Interfering with completions for modules
> * Being able to permission "push az provider" separately from "push any bicep module".

I apologize for being late to this conversation, but I want to emphasize that the concerns Anthony raised here are indeed valid. While interference with module completions may not pose a significant issue as we don't include provider metadata in `https://aks.ms/br-module-index-data`, users may still accidentally reference a provider in a module reference string, even without completions. The matter of permissions is also an aspect that should not be overlooked.

It appears to me that creating a new ACR, connecting it to MCR, and adding a deliciated MCR manifest for providers is a practical solution. The process should be relatively straightforward, as there is [shared infrastructure code](https://dev.azure.com/msazure/One/_git/Mgmt-Governance-Blueprint?path=/src/Ev2Deployment/ServiceGroupRoot/Templates/BicepRegistry.template.bicep) we can reuse to set up the ACR. Regarding provider paths, I recommend adopting a distinct naming convention such as `mcr.microsoft.com/bicep-providers/...` to avoid conflicts with public registry module paths.

Reply:
That works for me, @anthony-c-martin would you want for me to do that work before we merge this PR? Or can I simply change the default to bicep-providers and merge? And address the registry and publication to the registry (which happens in a different repo) separately?

Reply:
It works for me. When we're deploying the new ACR, please use the Premium SKU and enable zone redundancy.

Reply:
+1 - this PR doesn't need to blocked by it, we just need to make sure we're tracking it.